src/components/Spawn/editVolumeModal/transformer.ts

```diff
@@ -0,0 +1,23 @@
+import { diff } from "deep-object-diff";
+import { FormState } from "./types";
+
+export const formToGql = (
+  initialState: FormState,
+  formData: FormState,
+  volumeId: string
+) => {
+  const updatedFields: Partial<FormState> = diff(initialState, formData);
+  const { expirationDetails = {} as FormState["expirationDetails"], name } =
+    updatedFields;
+  return {
+    ...("noExpiration" in expirationDetails && {
+      noExpiration: expirationDetails.noExpiration,
+    }),
+    ...("expiration" in expirationDetails &&
+      !expirationDetails.noExpiration && {
+        expiration: new Date(expirationDetails.expiration),
+      }),
+    ...("name" in updatedFields && { name }),
+    volumeId,
+  };
```

Comment:
Is it possible to destructure these variables?
```suggestion
  const { expirationDetails = {} as FormState["expirationDetails"], name = "" } =
    updatedFields;
  const { noExpiration, expiration } = expirationDetails;
  
  return {
    ...(noExpiration && {
      noExpiration,
    }),
    ...(expiration &&
      !noExpiration && {
        expiration: new Date(expiration),
      }),
    ...(name && { name }),
    volumeId,
  };
```